# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1307375655 14400
#      Mon Jun 06 11:54:15 2011 -0400
# Branch stable
# Node ID 762d10c7727795445c74123077a94adbf0126bfb
# Parent  96180026afc536f2e06bf66907c73f303d0f90bd
maint: use libdir for .oct files, not libexecdir

From Orion Poplawski <orion@cora.nwra.com>.

* configure.ac (octlibdir): Use octave/$(version), not
octave-$(version).
(octfiledir): Use $(libdir), not $(libexecdir).
(localoctfiledir): Likewise.
(localapioctfiledir): Likewise.
(localveroctfiledir): Likewise.

* pkg.m: Use libdir instead of libexecdir as needed.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -120,33 +120,33 @@ OCTAVE_SET_DEFAULT(doc_cache_file, '$(oc
 OCTAVE_SET_DEFAULT(infofile, '$(infodir)/octave.info')
 OCTAVE_SET_DEFAULT(octincludedir, '$(includedir)/octave-$(version)/octave')
 OCTAVE_SET_DEFAULT(fcnfiledir, '$(datadir)/octave/$(version)/m')
 OCTAVE_SET_DEFAULT(localfcnfiledir, '$(datadir)/octave/site/m')
 OCTAVE_SET_DEFAULT(localapifcnfiledir,
   '$(datadir)/octave/site/$(api_version)/m')
 OCTAVE_SET_DEFAULT(localverfcnfiledir, '$(datadir)/octave/$(version)/site/m')
 OCTAVE_SET_DEFAULT(octetcdir, '$(datadir)/octave/$(version)/etc')
-OCTAVE_SET_DEFAULT(octlibdir, '$(libdir)/octave-$(version)')
+OCTAVE_SET_DEFAULT(octlibdir, '$(libdir)/octave/$(version)')
 OCTAVE_SET_DEFAULT(archlibdir,
   '$(libexecdir)/octave/$(version)/exec/$(canonical_host_type)')
 OCTAVE_SET_DEFAULT(localarchlibdir,
   '$(libexecdir)/octave/site/exec/$(canonical_host_type)')
 OCTAVE_SET_DEFAULT(localapiarchlibdir,
   '$(libexecdir)/octave/$(api_version)/site/exec/$(canonical_host_type)')
 OCTAVE_SET_DEFAULT(localverarchlibdir,
   '$(libexecdir)/octave/$(version)/site/exec/$(canonical_host_type)')
 OCTAVE_SET_DEFAULT(octfiledir,
-  '$(libexecdir)/octave/$(version)/oct/$(canonical_host_type)')
+  '$(libdir)/octave/$(version)/oct/$(canonical_host_type)')
 OCTAVE_SET_DEFAULT(localoctfiledir,
-  '$(libexecdir)/octave/site/oct/$(canonical_host_type)')
+  '$(libdir)/octave/site/oct/$(canonical_host_type)')
 OCTAVE_SET_DEFAULT(localapioctfiledir,
-  '$(libexecdir)/octave/site/oct/$(api_version)/$(canonical_host_type)')
+  '$(libdir)/octave/site/oct/$(api_version)/$(canonical_host_type)')
 OCTAVE_SET_DEFAULT(localveroctfiledir,
-  '$(libexecdir)/octave/$(version)/site/oct/$(canonical_host_type)')
+  '$(libdir)/octave/$(version)/site/oct/$(canonical_host_type)')
 OCTAVE_SET_DEFAULT(imagedir, '$(datadir)/octave/$(version)/imagelib')
 
 ### Find pkg-config executable (sets $PKG_CONFIG)
 
 PKG_PROG_PKG_CONFIG
 
 ### Make configure args available for other uses.
 
diff --git a/scripts/pkg/pkg.m b/scripts/pkg/pkg.m
--- a/scripts/pkg/pkg.m
+++ b/scripts/pkg/pkg.m
@@ -250,17 +250,17 @@ function [local_packages, global_package
                                      "octave_packages");
   mlock ();
 
   global_install = issuperuser ();
 
   if (prefix == -1)
     if (global_install)
       prefix = fullfile (OCTAVE_HOME (), "share", "octave", "packages");
-      archprefix = fullfile (octave_config_info ("libexecdir"),
+      archprefix = fullfile (octave_config_info ("libdir"),
                              "octave", "packages");
     else
       prefix = fullfile ("~", "octave");
       archprefix = prefix;
     endif
     prefix = tilde_expand (prefix);
     archprefix = tilde_expand (archprefix);
   endif
@@ -295,17 +295,17 @@ function [local_packages, global_package
         if (! user_prefix)
           prefix = tilde_expand (fullfile ("~", "octave"));
           archprefix = prefix;
         endif
       case "-global"
         global_install = true;
         if (! user_prefix)
           prefix = fullfile (OCTAVE_HOME (), "share", "octave", "packages");
-          archprefix = fullfile (octave_config_info ("libexecdir"),
+          archprefix = fullfile (octave_config_info ("libdir"),
                                  "octave", "packages");
         endif
       case available_actions
         if (strcmp (action, "none"))
           action = varargin{i};
         else
           error ("more than one action specified");
         endif
@@ -2214,24 +2214,24 @@ function emp = dirempty (nm, ign)
     endfor
     emp = true;
   else
     emp = true;
   endif
 endfunction
 
 function arch = getarch ()
-  persistent _arch = cstrcat (octave_config_info("canonical_host_type"), ...
-                             "-", octave_config_info("api_version"));
+  persistent _arch = cstrcat (octave_config_info ("canonical_host_type"),
+                              "-", octave_config_info ("api_version"));
   arch = _arch;
 endfunction
 
 function archprefix = getarchprefix (desc, global_install)
   if ((nargin == 2 && global_install) || (nargin < 2 && issuperuser ()))
-    archprefix = fullfile (octave_config_info ("libexecdir"), "octave",
+    archprefix = fullfile (octave_config_info ("libdir"), "octave",
                            "packages", cstrcat(desc.name, "-", desc.version));
   else
     archprefix = desc.dir;
   endif
 endfunction
 
 function archdir = getarchdir (desc)
   archdir = fullfile (desc.archprefix, getarch());
